udata/core/dataset/models.py

```diff
@@ -553,6 +554,31 @@ def clean(self):
         if self.frequency in LEGACY_FREQUENCIES:
             self.frequency = LEGACY_FREQUENCIES[self.frequency]
 
+        for key, value in self.extras.items():
```

Comment:
Since the validation logic is only in model and not form, when updating extras wrongly with api v1, api.validate() is successful but form.save() raises an error that is not properly returned (a 500 is returned).
I think we've answered this, but shouldn't we add this logic at the form level?